WEB-2 Start/stop node. Current node/blockchain status. Transfer tiles

nicosantangelo opened the pull request:
With the new additions the dashboard looks like this:

<img width="1395" alt="1" src="https://user-images.githubusercontent.com/692077/27971415-ed3621ee-6352-11e7-91d8-bfe0c302cb11.png">

<img width="1400" alt="2" src="https://user-images.githubusercontent.com/692077/27971372-c32c2f1a-6352-11e7-886f-6a5b263fbdff.png">

While the dashboard is functional in it's current form, it's lacking a couple of features which will make the UX a lot better.

**For example**

- We don't have a good way to send notifications in case a request fails (or succeeds)
- The information is static, meaning that the mining speed and tile list will stay the same until a page refresh
- If the user has **a lot** of tiles, it could choke the UI

Let me know if you want me to address any (or all) of those in this PR. I think that the code it's at a good point to be transformed to use a framework (it has <200 lines of code and little HTML).

I was thinking of using React (with redux I think) to do so, but I'm open to suggestions!

That's all!

nicosantangelo committed:
WEB-2 Introduce templates and the current miner and blockchain info

nicosantangelo committed:
WEB-2 Start/stop nodes

nicosantangelo committed:
WEB-2 Transfer tiles to an address in batch

martriay commented on browser/js/main.js:
```diff
@@ -43,8 +155,8 @@
-      .catch(catchError)
-    })
+      .catch(catchError);
+    });
```
😢

nicosantangelo replied:
Keeping with the style of the hole codebase. I should add something to enforce it, because I keep forgetting to add them :/

maraoz commented:
Nice work @NicoSantangelo !

martriay commented:
*Feedback*:
- I'd set the on/off mining as a toggle and replace the informative sentence into an easy-to-read table
- I also find this sentence not quite needed:
```
Welcome. From this page you will be able to manage your node and see your mined tiles. You can send rpc calls to the node and see the result here
```
- As there is no more hash rate next to the `Tiles` title, i'd move the tile count next to it

NicoSantangelo commented:
> - It's not "start/stop node" but "start/stop miner"
👍 

> - Actually, I'd set the on/off mining as a toggle and replace the informative sentence into an easy-to-read table

By a toggle you mean a native radio element or a styled toggle button like:
![image](https://user-images.githubusercontent.com/692077/27974306-bdceda6c-635d-11e7-9d15-a462a11225f5.png)
 

> - I also find this sentence not quite needed:

👍 

> - As there is no more hash rate next to the `Tiles` title, i'd move the tile count next to it

Because we now show the 
- I can't find the `transfer tile` button

martriay commented:
@NicoSantangelo I was thinking in that same toggle you pasted.
> Could it be a cache problem?

Mmmh, tried pulling the repo, re-building the miner andning the web ui on incognito but i'm still getting this:
![image](https://user-images.githubusercontent.com/1703831/27974354-166d0bfa-6334-11e7-9bf9-c70a42f95212.png)

nicosantangelo committed:
WEB-2 Remove bad rebase

NicoSantangelo commented:
@martriay I managed to reproduce it by removing-refetching the branch. There was a problem with the last rebase, which left two functions with the same name on `main.js`. The last one on the file was outdated and overrode the newer one.

It should work now!

eordano reviewed (approved):
Looks good to me

martriay commented:
@NicoSantangelo it works now! But the main feedback remains:
- Toggle and wording (`miner` instead of `node`)
- Table instead of long sentence for mining specs

eordano commented:
I guess this is more of an epic, but this is what I would use for this admin interface:

https://blackrockdigital.github.io/startbootstrap-sb-admin-2/pages/index.html

nicosantangelo committed:
WEB-2 node to miner

NicoSantangelo commented:
I've updated the missnamed node action to miner. Let me know if you want me to try to add the table and toggle @martriay suggested or if we're going to redesign the dashboard (in another issue) like @eordano proposed.

Also, just in case you missed it, we need to discuss the points mentioned on the first post eventually:

>- We don't have a good way to send notifications in case a request fails (or succeeds)
> - The information is static, meaning that the mining speed and tile list will stay the same until a page refresh
> - If the user has **a lot** of tiles, it could choke the UI
>
> (...)
>
> I think that the code it's at a good point to be transformed to use a framework (it has <200 lines of code and little HTML). I was thinking of using React (with redux I think) to do so, but I'm open to suggestions!

martriay commented:
- +1 to react, go for it
- we don't even have a mock of the design, so please go on with the toggle and table
- static info, edge cases (too many tiles) and non-response is ok in this instance

NicoSantangelo commented:
@martriay I added the table and changed the node references to miner. I'm having troubles placing the toggle thought. 

First I came up with this (current commited version)

<img width="1376" alt="screen shot 2017-07-11 at 11 48 25" src="https://user-images.githubusercontent.com/692077/28063610-ef785f4e-6631-11e7-9342-b8078b184557.png">

From there, adding the toggle it's not that easy, here are some options:
<img width="1423" alt="screen shot 2017-07-11 at 12 08 00" src="https://user-images.githubusercontent.com/692077/28063627-00500920-6632-11e7-9f74-169951d92bda.png">
<img width="1420" alt="screen shot 2017-07-11 at 12 10 06" src="https://user-images.githubusercontent.com/692077/28063628-0056ee7a-6632-11e7-99ca-31d7b117a6c0.png">
<img width="1380" alt="screen shot 2017-07-11 at 12 21 36" src="https://user-images.githubusercontent.com/692077/28064047-86dde060-6633-11e7-93a2-df0727b98480.png">

Let me know if you like any of them or if you have any other ideas and I'll update the branch :)

nicosantangelo committed:
WEB-2 Update the dashboard to use 4 sections and a toggle

nicosantangelo committed:
WEB-2 Pseudo bootstrap + better separation between boxes

nicosantangelo committed:
WEB-2 Remove unnecessary loading items. Finish styling
